compiler/optimizations/gpuTransforms.cpp

```diff
@@ -1003,6 +1003,53 @@ struct KernelActual {
   int8_t kind;  // assigned to one or more values of GpuArgKind or'd together
 };
 
+enum ReductionKind { SUM, MIN, MAX, UNSUPPORTED };
```

Comment:
I think they are all doable. `&&` doesn't have a function we can use directly, but CUB supports custom reductions. We need to implement our own function for `&&`, which I expect to be straightforward, but out of my comfort zone for this PR.

`minloc`/`maxloc` are also possible -- I think they're call argmin/argmax. We'll need to wire things a bit differently. For example, instead of having a single interim buffer for per-block reduction results, we need two of those where the second one would have to store the corresponding indices. Again, doable, but not _that_ simple.

I can record these in an issue as followups.